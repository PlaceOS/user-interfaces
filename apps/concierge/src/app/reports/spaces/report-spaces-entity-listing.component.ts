import { Component, Input } from '@angular/core';
import { downloadFile, jsonToCsv } from '@placeos/common';
import { formatDuration } from 'date-fns';
import { combineLatest } from 'rxjs';
import { debounceTime, map, shareReplay, take } from 'rxjs/operators';
import { ReportsStateService } from '../reports-state.service';

@Component({
    selector: 'report-spaces-entity-listing',
    template: `
        <div
            class="m-4 rounded bg-base-100 border border-base-200 overflow-hidden"
        >
            <div class="border-b border-base-200 px-4 py-2 flex items-center">
                <h3 class="font-bold text-xl flex-1">Visitor Entities</h3>
                <button icon matRipple (click)="download()" *ngIf="!print">
                    <app-icon>download</app-icon>
                </button>
            </div>
            <simple-table
                class="w-full block text-sm"
                [data]="entity_list"
                [columns]="[
                    { key: 'name', name: 'Name' },
                    { key: 'booking_count', name: 'Bookings' },
                    { key: 'avg_attendees', name: 'Avg. Invitees per Booking' },
                    { key: 'total_time', name: 'Total Booked Time' },
                    { key: 'no_shows', name: 'No Shows' },
                ]"
                [sortable]="true"
                [page_size]="print ? 0 : 10"
                empty_message="No events for selected period"
            ></simple-table>
        </div>
    `,
    styles: [``],
})
export class ReportSpacesEntityListingComponent {
    @Input() public print: boolean = false;

    public readonly entity_list = combineLatest([this._reports.stats]).pipe(
        debounceTime(300),
        map(([stats]) => {
            let list = [];
            for (const booking of stats.events) {
                const entity = booking.extension_data?.visitor_entity;
                if (!entity) continue;
                const capacity = Math.max(
                    booking.resources.reduce((c, s) => c + s.capacity, 0) || 1,
                    1,
                );
                let details = list.find(
                    (_) => _.id?.toLowerCase() === entity.toLowerCase(),
                );
                if (!details) {
                    details = {
                        id: entity,
                        name: entity,
                        capacity,
                        booking_count: 0,
                        attendees: 0,
                        avg_attendees: 0,
                        no_shows: 0,
                        occupancy: 0,
                        total_time: 0,
                    };
                    list.push(details);
                }
                if (booking.extension_data?.people_count?.max === 0) {
                    details.no_shows += 1;
                }
                details.booking_count += 1;
                details.attendees += booking.attendees.length;
                details.total_time += booking.duration || 15;
            }
            for (const space of list) {
                space.avg_attendees =
                    Math.floor((space.attendees / space.booking_count) * 100) /
                    100;
                space.occupancy =
                    Math.floor((space.avg_attendees / space.capacity) * 100) /
                    100;
                space.total_time = formatDuration({
                    hours: Math.floor(space.total_time / 60),
                    minutes: space.total_time % 60,
                });
            }
            return list;
        }),
        shareReplay(1),
    );

    public readonly download = async () => {
        const data = await this.entity_list.pipe(take(1)).toPromise();
        for (const item of data) {
            delete item.attendance;
            delete item.avg_attendance;
            delete item.min_attendance;
            delete item.max_attendance;
            delete item.occupancy;
        }
        downloadFile('report-space-attendee-usage.csv', jsonToCsv(data));
    };

    constructor(private _reports: ReportsStateService) {}
}
