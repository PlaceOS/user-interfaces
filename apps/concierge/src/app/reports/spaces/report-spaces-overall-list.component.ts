import { Component, Input } from '@angular/core';
import { downloadFile, jsonToCsv } from '@placeos/common';
import { format } from 'date-fns';
import { take } from 'rxjs/operators';

import { ReportsStateService } from '../reports-state.service';

@Component({
    selector: 'report-spaces-overall-list',
    template: `
        <div class="w-full">
            <div
                class="m-4 rounded bg-base-100 border border-base-200 overflow-hidden"
            >
                <div class="border-b border-base-200 p-4 flex items-center">
                    <h3 class="font-bold text-xl flex-1">Daily Utilisation</h3>
                    <button icon matRipple *ngIf="!print" (click)="download()">
                        <app-icon>download</app-icon>
                    </button>
                </div>
                <simple-table
                    class="w-full block text-sm"
                    [data]="day_list"
                    [columns]="[
                        { key: 'date', name: 'Date', content: date_template },
                        { key: 'approved', name: 'Approved Bookings' },
                        { key: 'count', name: 'Total Requests' },
                        {
                            key: 'utilisation',
                            name: 'Utilisation',
                            content: percent_template
                        }
                    ]"
                    [sortable]="true"
                    [page_size]="print ? 0 : 10"
                    empty_message="No events for selected period"
                ></simple-table>
                <ng-template #date_template let-data="data">
                    <div class="p-4">{{ data | date: 'mediumDate' }}</div>
                </ng-template>
                <ng-template #percent_template let-data="data">
                    <div class="p-4">{{ data || '0' }}%</div>
                </ng-template>
            </div>
        </div>
    `,
})
export class ReportSpacesOverallListComponent {
    @Input() public print: boolean = false;

    public readonly day_list = this._state.day_list;

    public readonly download = async () => {
        let data = await this.day_list.pipe(take(1)).toPromise();
        data = data.map((d) => ({
            ...d,
            date: format(d.date, 'MMMM d, y(EEE)'),
        }));
        downloadFile('spaces-usage.csv', jsonToCsv(data));
    };

    constructor(private _state: ReportsStateService) {}
}
