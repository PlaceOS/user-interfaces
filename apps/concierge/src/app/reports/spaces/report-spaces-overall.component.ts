import { Component } from '@angular/core';
import { formatDuration } from 'date-fns';
import { map } from 'rxjs/operators';
import { ReportsStateService } from '../reports-state.service';

@Component({
    selector: 'report-spaces-overall',
    template: `
    <div
            class="m-4 p-4 rounded bg-white shadow flex justify-center items-center space-x-2"
        >
            <div class="flex flex-col items-center flex-1">
                <h3>Total Bookings</h3>
                <p>{{ (total_count | async) || 0 }}</p>
            </div>
            <div class="flex flex-col items-center flex-1">
                <h3>Average Bookings Length</h3>
                <p>{{ (avg_length | async) || 'None' }}</p>
            </div>
        </div>
        <div
            class="m-4 p-4 rounded bg-white shadow flex justify-center items-center space-x-2"
        >
            <div class="flex-1 flex flex-col items-center justify-center">
                <h3>Overall Efficiency</h3>
                <p>{{ (efficiency | async) || 0 }}%</p>
            </div>
            <div class="flex-1 flex flex-col items-center justify-center">
                <h3>Occupancy</h3>
                <p>{{ (occupancy | async) || 0 }}%</p>
            </div>
            <div class="flex-1 flex flex-col items-center justify-center">
                <h3>Total Attendees</h3>
                <p>{{ (total_users | async) || 0 }}</p>
            </div>
            <div class="flex-1 flex flex-col items-center justify-center">
                <h3>Total Capacity</h3>
                <p>{{ (total_capacity | async) || 0 }}</p>
            </div>
            <div class="flex-1 flex flex-col items-center justify-center">
                <h3>Utilisation</h3>
                <p>{{ (utilisation | async) || 0 }}%</p>
            </div>
        </div>`,
    styles: [`
        h3 {
            font-weight: 500;
        }

        p:not([simple]) {
            font-size: 1.75rem;
        }
    `],
})
export class ReportSpacesOverallComponent {
    public readonly total_count = this._state.stats.pipe(
        map((i) => i.count || 0)
    );
    public readonly avg_length = this._state.stats.pipe(
        map((i) => formatDuration({ minutes: i.avg_length || 0 }))
    );
    public readonly efficiency = this._state.stats.pipe(
        map((i) => ((i.efficiency || 0) * 100).toFixed(1))
    );
    public readonly total_users = this._state.stats.pipe(
        map((i) => i.total_users || 0)
    );
    public readonly total_capacity = this._state.stats.pipe(
        map((i) => i.total_capacity || 0)
    );
    public readonly occupancy = this._state.stats.pipe(
        map((i) => ((i.occupancy || 0) * 100).toFixed(1))
    );
    public readonly utilisation = this._state.stats.pipe(
        map((i) => ((i.utilisation || 0) * 100).toFixed(1))
    );

    constructor(private _state: ReportsStateService) {}
}
