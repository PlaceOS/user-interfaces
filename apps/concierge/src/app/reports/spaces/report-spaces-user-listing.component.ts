import { Component, Input } from '@angular/core';
import { downloadFile, jsonToCsv } from '@placeos/common';
import { User } from '@placeos/users';
import { formatDuration } from 'date-fns';
import { combineLatest } from 'rxjs';
import { debounceTime, map, shareReplay, take } from 'rxjs/operators';
import { ReportsStateService } from '../reports-state.service';

@Component({
    selector: 'report-spaces-user-listing',
    template: `
        <div
            class="m-4 rounded bg-base-100 border border-base-200 overflow-hidden"
        >
            <div class="border-b border-base-200 px-4 py-2 flex items-center">
                <h3 class="font-bold text-xl flex-1">Meeting Organisers</h3>
                <button icon matRipple (click)="download()" *ngIf="!print">
                    <app-icon>download</app-icon>
                </button>
            </div>
            <simple-table
                class="w-full block text-sm"
                [data]="user_list"
                [columns]="[
                    { key: 'name', name: 'Name' },
                    { key: 'booking_count', name: 'Bookings' },
                    { key: 'avg_attendees', name: 'Avg. Invitees per Booking' },
                    { key: 'total_time', name: 'Total Booked Time' },
                    { key: 'no_shows', name: 'No Shows' }
                ]"
                [sortable]="true"
                [page_size]="print ? 0 : 10"
                empty_message="No events for selected period"
            ></simple-table>
        </div>
    `,
    styles: [``],
})
export class ReportSpacesUserListingComponent {
    @Input() public print: boolean = false;

    public readonly user_list = combineLatest([this._reports.stats]).pipe(
        debounceTime(300),
        map(([stats]) => {
            let list = [];
            for (const booking of stats.events) {
                const host = booking.attendees?.find(
                    (_) =>
                        _.email === booking.extension_data?.host_override ||
                        _.email === booking.host
                );
                if (!host) continue;
                const capacity = Math.max(
                    booking.resources.reduce((c, s) => c + s.capacity, 0) || 1,
                    1
                );
                let details = list.find(
                    (_) => _.id?.toLowerCase() === host.email.toLowerCase()
                );
                if (!details) {
                    details = {
                        id: host.email,
                        name: host.name,
                        capacity,
                        booking_count: 0,
                        attendees: 0,
                        avg_attendees: 0,
                        no_shows: 0,
                        occupancy: 0,
                        total_time: 0,
                    };
                    list.push(details);
                }
                if (booking.extension_data?.people_count?.max === 0) {
                    details.no_shows += 1;
                }
                details.booking_count += 1;
                details.attendees += booking.attendees.length;
                details.total_time += booking.duration || 15;
            }
            for (const space of list) {
                space.avg_attendees =
                    Math.floor((space.attendees / space.booking_count) * 100) /
                    100;
                space.occupancy =
                    Math.floor((space.avg_attendees / space.capacity) * 100) /
                    100;
                space.total_time = formatDuration({
                    hours: Math.floor(space.total_time / 60),
                    minutes: space.total_time % 60,
                });
            }
            return list;
        }),
        shareReplay(1)
    );

    public readonly download = async () => {
        const data = await this.user_list.pipe(take(1)).toPromise();
        for (const item of data) {
            delete item.attendance;
            delete item.avg_attendance;
            delete item.min_attendance;
            delete item.max_attendance;
            delete item.occupancy;
        }
        downloadFile('report-space-attendee-usage.csv', jsonToCsv(data));
    };

    constructor(private _reports: ReportsStateService) {}
}
