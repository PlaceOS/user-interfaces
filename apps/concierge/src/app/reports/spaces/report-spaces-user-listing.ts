import { Component } from '@angular/core';
import { downloadFile, jsonToCsv } from '@placeos/common';
import { User } from '@placeos/users';
import { combineLatest } from 'rxjs';
import { debounceTime, map, shareReplay, take } from 'rxjs/operators';
import { ReportsStateService } from '../reports-state.service';

@Component({
    selector: 'report-spaces-user-listing',
    template: `
        <div class="m-4 rounded bg-white shadow overflow-hidden">
            <div class="border-b border-gray-300 px-4 py-2 flex items-center">
                <h3 class="font-bold text-xl flex-1">Staff Utilisation</h3>
                <button mat-icon-button (click)="download()">
                    <app-icon>download</app-icon>
                </button>
            </div>
            <custom-table
                [dataSource]="user_list"
                [pagination]="true"
                [columns]="[
                    'name',
                    'capacity',
                    'count',
                    'avg_attendees',
                    'occupancy'
                ]"
                [display_column]="[
                    'Name',
                    'Capacity',
                    'Meeting Count',
                    'Avg. Attendees',
                    'Occupancy'
                ]"
                [column_size]="['flex']"
            ></custom-table>
        </div>
    `,
    styles: [``],
})
export class ReportSpacesUserListing {
    public readonly user_list = combineLatest([this._reports.stats]).pipe(
        debounceTime(300),
        map(([stats]) => {
            let list = [];
            for (const booking of stats.events) {
                const attendees: User[] = booking.attendees || [];
                const capacity =
                    booking.resources.reduce((c, s) => c + s.capacity, 0) || 1;
                for (const user of attendees) {
                    if (!list.find((_) => _.email === user.email)) {
                        list.push({
                            id: user.email,
                            name: user.name,
                            capacity,
                            count: 0,
                            attendees: 0,
                            avg_attendees: 0,
                            occupancy: 0,
                        });
                    }
                    const details = list.find((_) => _.id === user.email);
                    details.count += 1;
                    details.attendees += booking.attendees.length;
                }
            }
            for (const space of list) {
                space.avg_attendees = space.attendees / space.count;
                space.occupancy = space.avg_attendees / space.capacity;
            }
            return list;
        }),
        shareReplay(1)
    );

    public readonly download = async () => {
        const data = await this.user_list.pipe(take(1)).toPromise();
        downloadFile('report-space-attendee-usage.csv', jsonToCsv(data));
    };

    constructor(private _reports: ReportsStateService) {}
}
