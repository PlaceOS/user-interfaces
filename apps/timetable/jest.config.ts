/* eslint-disable */
export default {
    displayName: 'timetable',
    preset: '../../jest.preset.js',
    setupFilesAfterEnv: ['<rootDir>/src/test-setup.ts'],
    globals: {},
    coverageDirectory: '../../coverage/apps/timetable',
    transform: {
        '^.+\\.(ts|mjs|js|html)$': [
            'jest-preset-angular',
            {
                tsconfig: '<rootDir>/tsconfig.spec.json',
                stringifyContentPathRegex: '\\.(html|svg)$',
            },
        ],
    },
    transformIgnorePatterns: ['node_modules/(?!.*\\.mjs$)'],
    moduleNameMapper: {
        '^quill$': 'node_modules/quill/dist/quill.js',
    },
    snapshotSerializers: [
        'jest-preset-angular/build/serializers/no-ng-attributes',
        'jest-preset-angular/build/serializers/ng-snapshot',
        'jest-preset-angular/build/serializers/html-comment',
    ],
};
