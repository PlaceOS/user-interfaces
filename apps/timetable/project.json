{
    "name": "timetable",
    "$schema": "../../node_modules/nx/schemas/project-schema.json",
    "projectType": "application",
    "sourceRoot": "apps/timetable/src",
    "prefix": "placeos",
    "targets": {
        "build": {
            "executor": "@angular-devkit/build-angular:browser",
            "outputs": ["{options.outputPath}"],
            "options": {
                "outputPath": "dist/apps/timetable",
                "index": "apps/timetable/src/index.html",
                "main": "apps/timetable/src/main.ts",
                "polyfills": "apps/timetable/src/polyfills.ts",
                "tsConfig": "apps/timetable/tsconfig.app.json",
                "baseHref": ".",
                "assets": [
                    "apps/timetable/src/favicon.ico",
                    "apps/timetable/src/assets",
                    {
                        "glob": "**",
                        "input": "shared/assets",
                        "output": "/assets/"
                    },
                    {
                        "glob": "oauth-resp.html",
                        "input": "node_modules/@placeos/ts-client/dist/",
                        "output": "/"
                    },
                    "apps/timetable/src/manifest.webmanifest"
                ],
                "styles": [
                    "./node_modules/@angular/material/prebuilt-themes/indigo-pink.css",
                    "apps/timetable/src/styles.css",
                    "shared/styles/application.scss",
                    "shared/styles/custom-element.scss",
                    "shared/styles/native-element.scss",
                    "shared/styles/utility.scss"
                ],
                "scripts": [],
                "serviceWorker": true,
                "ngswConfigPath": "apps/timetable/ngsw-config.json"
            },
            "configurations": {
                "staging": {
                    "optimization": true,
                    "buildOptimizer": true,
                    "fileReplacements": [
                        {
                            "replace": "apps/timetable/src/environments/environment.ts",
                            "with": "apps/timetable/src/environments/environment.prod.ts"
                        },
                        {
                            "replace": "libs/common/src/lib/settings.ts",
                            "with": "apps/timetable/src/environments/settings.ts"
                        }
                    ],
                    "serviceWorker": true,
                    "ngswConfigPath": "apps/timetable/ngsw-config.json"
                },
                "production": {
                    "budgets": [
                        {
                            "type": "initial",
                            "maximumWarning": "1mb",
                            "maximumError": "4mb"
                        },
                        {
                            "type": "anyComponentStyle",
                            "maximumWarning": "2kb",
                            "maximumError": "4kb"
                        }
                    ],
                    "fileReplacements": [
                        {
                            "replace": "libs/mocks/src/index.ts",
                            "with": "libs/mocks/src/empty.ts"
                        },
                        {
                            "replace": "apps/timetable/src/environments/environment.ts",
                            "with": "apps/timetable/src/environments/environment.prod.ts"
                        },
                        {
                            "replace": "libs/common/src/lib/settings.ts",
                            "with": "apps/timetable/src/environments/settings.ts"
                        }
                    ],
                    "outputHashing": "all"
                },
                "development": {
                    "buildOptimizer": false,
                    "optimization": false,
                    "vendorChunk": true,
                    "extractLicenses": false,
                    "sourceMap": true,
                    "namedChunks": true
                }
            },
            "defaultConfiguration": "production"
        },
        "serve": {
            "executor": "@angular-devkit/build-angular:dev-server",
            "configurations": {
                "production": {
                    "buildTarget": "timetable:build:production"
                },
                "development": {
                    "buildTarget": "timetable:build:development"
                }
            },
            "defaultConfiguration": "development"
        },
        "extract-i18n": {
            "executor": "@angular-devkit/build-angular:extract-i18n",
            "options": {
                "buildTarget": "timetable:build"
            }
        },
        "lint": {
            "executor": "@nx/eslint:lint"
        },
        "test": {
            "executor": "@nx/jest:jest",
            "outputs": ["{workspaceRoot}/coverage/apps/timetable"],
            "options": {
                "jestConfig": "apps/timetable/jest.config.ts"
            }
        }
    },
    "tags": []
}
