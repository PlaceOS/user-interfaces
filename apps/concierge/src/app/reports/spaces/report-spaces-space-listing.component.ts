import { Component } from '@angular/core';
import { downloadFile, jsonToCsv } from '@placeos/common';
import { Space } from '@placeos/spaces';
import { differenceInDays } from 'date-fns';
import { combineLatest } from 'rxjs';
import { debounceTime, map, take } from 'rxjs/operators';
import { ReportsStateService } from '../reports-state.service';

@Component({
    selector: 'report-spaces-space-listing',
    template: `
        <div class="m-4 rounded bg-white shadow overflow-hidden">
            <div class="border-b border-gray-300 px-4 py-2 flex items-center">
                <h3 class="font-bold text-xl flex-1">Space Utilisation</h3>
                <button mat-icon-button (click)="download()">
                    <app-icon>download</app-icon>
                </button>
            </div>
            <custom-table
                [dataSource]="space_list"
                [pagination]="true"
                [columns]="[
                    'name',
                    'capacity',
                    'count',
                    'utilisation',
                    'avg_attendees',
                    'occupancy'
                ]"
                [display_column]="[
                    'Name',
                    'Capacity',
                    'Meeting Count',
                    'Utilisation',
                    'Avg. Attendees',
                    'Occupancy'
                ]"
                [column_size]="['flex']"
            ></custom-table>
        </div>
    `,
    styles: [``],
})
export class ReportSpacesSpaceListing {

    public readonly space_list = combineLatest([
        this._reports.stats,
        this._reports.options,
    ]).pipe(
        debounceTime(300),
        map(([stats, { start, end }]) => {
            let list = [];
            for (const booking of stats.events) {
                const resources: Space[] = booking.resources || [];
                for (const space of resources) {
                    if (!list.find((_) => _.id === space.id)) {
                        list.push({
                            id: space.id,
                            name: space.display_name || space.name,
                            capacity: space.capacity || 1,
                            count: 0,
                            attendees: 0,
                            avg_attendees: 0,
                            usage: 0,
                            utilisation: 0,
                            occupancy: 0,
                        });
                    }
                    const details = list.find((_) => _.id === space.id);
                    details.count += 1;
                    details.usage += booking.duration;
                    details.attendees += booking.attendees.length;
                }
            }
            const period_in_days = differenceInDays(end, start);
            for (const space of list) {
                space.avg_attendees = space.attendees / space.count;
                space.utilisation = space.usage / 60 / 8 / period_in_days;
                space.occupancy = space.avg_attendees / space.capacity;
            }
            return list;
        })
    );

    public readonly download = async () => {
        const data = await this.space_list.pipe(take(1)).toPromise();
        downloadFile('report-spaces-usage.csv', jsonToCsv(data));
    };

    constructor(private _reports: ReportsStateService) {}
}
