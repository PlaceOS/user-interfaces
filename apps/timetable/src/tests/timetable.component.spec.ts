import { createRoutingFactory, SpectatorRouting } from '@ngneat/spectator/jest';
import { SettingsService } from '@placeos/common';
import { SpacesService } from '@placeos/spaces';
import { of } from 'rxjs';
import { MockComponent } from 'ng-mocks';

import { AppTimetableComponent } from '../app/timetable.component';
import { SpaceTimetableComponent } from '../app/space-timetable.component';

describe('AppTimetableComponent', () => {
    let spectator: SpectatorRouting<AppTimetableComponent>;
    const createComponent = createRoutingFactory({
        component: AppTimetableComponent,
        providers: [
            { provide: SettingsService, useValue: { get: jest.fn() } },
            {
                provide: SpacesService,
                useValue: { initialised: of(true), find: jest.fn() },
            },
        ],
        declarations: [MockComponent(SpaceTimetableComponent)],
    });

    beforeEach(() => (spectator = createComponent()));

    it('should create component', () => {
        expect(spectator.component).toBeTruthy();
    });
});
