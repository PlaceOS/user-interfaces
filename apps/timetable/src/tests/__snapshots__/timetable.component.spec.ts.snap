// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTimetableComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="absolute inset-0 flex flex-col"
  >
    <div
      class="bg-secondary w-full shadow z-20 p-2 h-16 flex items-center relative"
      topbar=""
    >
      <img
        class="h-12"
        src=""
      />
      <div
        class="flex-1"
      />
      <div
        class="text-white p-2 text-xl"
      >
        <span>
          Jan 1, 1970
        </span>
        <span
          class="mx-2"
        >
          â€¢
        </span>
        <span
          class="ml-1"
        >
          10:00 AM
        </span>
      </div>
    </div>
    <div
      class="flex items-center overflow-auto h-1/2 flex-1 w-full bg-[#424242] z-10 relative flex-wrap"
    >
      <div
        class="sticky left-0 min-h-full w-16 min-w-[4rem] border-r border-white/50 bg-[#212121] flex flex-col z-20"
      >
        <div
          class="w-full min-h-[3rem] border-b border-white/50"
        />
        <div
          class="w-full flex flex-col flex-1 h-1/2 relative"
        >
          <div
            class="absolute left-0 w-screen h-[2px] bg-primary -translate-y-1/2 z-20"
            now=""
            style="top: 41.66666666666667%;"
          >
            <div
              class="arrow absolute left-0 top-0 -translate-y-1/2"
            />
          </div>
          <div
            class="w-full flex-1 min-h-[2rem] border-b border-white/50 relative z-10"
            hour=""
          >
            <div
              class="text-white bg-[#212121] w-8 text-center absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
              text=""
            >
               12 
            </div>
            <div
              class="absolute top-1/2 inset-x-0 w-full border-b border-white/50"
            />
          </div>
          <div
            class="w-full flex-1 min-h-[2rem] border-b border-white/50 relative z-10"
            hour=""
          >
            <div
              class="text-white bg-[#212121] w-8 text-center absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
              text=""
            >
               1 
            </div>
            <div
              class="absolute top-1/2 inset-x-0 w-full border-b border-white/50"
            />
          </div>
          <div
            class="w-full flex-1 min-h-[2rem] border-b border-white/50 relative z-10"
            hour=""
          >
            <div
              class="text-white bg-[#212121] w-8 text-center absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
              text=""
            >
               2 
            </div>
            <div
              class="absolute top-1/2 inset-x-0 w-full border-b border-white/50"
            />
          </div>
          <div
            class="w-full flex-1 min-h-[2rem] border-b border-white/50 relative z-10"
            hour=""
          >
            <div
              class="text-white bg-[#212121] w-8 text-center absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
              text=""
            >
               3 
            </div>
            <div
              class="absolute top-1/2 inset-x-0 w-full border-b border-white/50"
            />
          </div>
          <div
            class="w-full flex-1 min-h-[2rem] border-b border-white/50 relative z-10"
            hour=""
          >
            <div
              class="text-white bg-[#212121] w-8 text-center absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
              text=""
            >
               4 
            </div>
            <div
              class="absolute top-1/2 inset-x-0 w-full border-b border-white/50"
            />
          </div>
          <div
            class="w-full flex-1 min-h-[2rem] border-b border-white/50 relative z-10"
            hour=""
          >
            <div
              class="text-white bg-[#212121] w-8 text-center absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
              text=""
            >
               5 
            </div>
            <div
              class="absolute top-1/2 inset-x-0 w-full border-b border-white/50"
            />
          </div>
          <div
            class="w-full flex-1 min-h-[2rem] border-b border-white/50 relative z-10"
            hour=""
          >
            <div
              class="text-white bg-[#212121] w-8 text-center absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
              text=""
            >
               6 
            </div>
            <div
              class="absolute top-1/2 inset-x-0 w-full border-b border-white/50"
            />
          </div>
          <div
            class="w-full flex-1 min-h-[2rem] border-b border-white/50 relative z-10"
            hour=""
          >
            <div
              class="text-white bg-[#212121] w-8 text-center absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
              text=""
            >
               7 
            </div>
            <div
              class="absolute top-1/2 inset-x-0 w-full border-b border-white/50"
            />
          </div>
          <div
            class="w-full flex-1 min-h-[2rem] border-b border-white/50 relative z-10"
            hour=""
          >
            <div
              class="text-white bg-[#212121] w-8 text-center absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
              text=""
            >
               8 
            </div>
            <div
              class="absolute top-1/2 inset-x-0 w-full border-b border-white/50"
            />
          </div>
          <div
            class="w-full flex-1 min-h-[2rem] border-b border-white/50 relative z-10"
            hour=""
          >
            <div
              class="text-white bg-[#212121] w-8 text-center absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
              text=""
            >
               9 
            </div>
            <div
              class="absolute top-1/2 inset-x-0 w-full border-b border-white/50"
            />
          </div>
          <div
            class="w-full flex-1 min-h-[2rem] border-b border-white/50 relative z-10"
            hour=""
          >
            <div
              class="text-white bg-[#212121] w-8 text-center absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
              text=""
            >
               10 
            </div>
            <div
              class="absolute top-1/2 inset-x-0 w-full border-b border-white/50"
            />
          </div>
          <div
            class="w-full flex-1 min-h-[2rem] border-b border-white/50 relative z-10"
            hour=""
          >
            <div
              class="text-white bg-[#212121] w-8 text-center absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
              text=""
            >
               11 
            </div>
            <div
              class="absolute top-1/2 inset-x-0 w-full border-b border-white/50"
            />
          </div>
          <div
            class="w-full flex-1 min-h-[2rem] border-b border-white/50 relative z-10"
            hour=""
          >
            <div
              class="text-white bg-[#212121] w-8 text-center absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
              text=""
            >
               12 
            </div>
            <div
              class="absolute top-1/2 inset-x-0 w-full border-b border-white/50"
            />
          </div>
          <div
            class="w-full flex-1 min-h-[2rem] border-b border-white/50 relative z-10"
            hour=""
          >
            <div
              class="text-white bg-[#212121] w-8 text-center absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
              text=""
            >
               1 
            </div>
            <div
              class="absolute top-1/2 inset-x-0 w-full border-b border-white/50"
            />
          </div>
          <div
            class="w-full flex-1 min-h-[2rem] border-b border-white/50 relative z-10"
            hour=""
          >
            <div
              class="text-white bg-[#212121] w-8 text-center absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
              text=""
            >
               2 
            </div>
            <div
              class="absolute top-1/2 inset-x-0 w-full border-b border-white/50"
            />
          </div>
          <div
            class="w-full flex-1 min-h-[2rem] border-b border-white/50 relative z-10"
            hour=""
          >
            <div
              class="text-white bg-[#212121] w-8 text-center absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
              text=""
            >
               3 
            </div>
            <div
              class="absolute top-1/2 inset-x-0 w-full border-b border-white/50"
            />
          </div>
          <div
            class="w-full flex-1 min-h-[2rem] border-b border-white/50 relative z-10"
            hour=""
          >
            <div
              class="text-white bg-[#212121] w-8 text-center absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
              text=""
            >
               4 
            </div>
            <div
              class="absolute top-1/2 inset-x-0 w-full border-b border-white/50"
            />
          </div>
          <div
            class="w-full flex-1 min-h-[2rem] border-b border-white/50 relative z-10"
            hour=""
          >
            <div
              class="text-white bg-[#212121] w-8 text-center absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
              text=""
            >
               5 
            </div>
            <div
              class="absolute top-1/2 inset-x-0 w-full border-b border-white/50"
            />
          </div>
          <div
            class="w-full flex-1 min-h-[2rem] border-b border-white/50 relative z-10"
            hour=""
          >
            <div
              class="text-white bg-[#212121] w-8 text-center absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
              text=""
            >
               6 
            </div>
            <div
              class="absolute top-1/2 inset-x-0 w-full border-b border-white/50"
            />
          </div>
          <div
            class="w-full flex-1 min-h-[2rem] border-b border-white/50 relative z-10"
            hour=""
          >
            <div
              class="text-white bg-[#212121] w-8 text-center absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
              text=""
            >
               7 
            </div>
            <div
              class="absolute top-1/2 inset-x-0 w-full border-b border-white/50"
            />
          </div>
          <div
            class="w-full flex-1 min-h-[2rem] border-b border-white/50 relative z-10"
            hour=""
          >
            <div
              class="text-white bg-[#212121] w-8 text-center absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
              text=""
            >
               8 
            </div>
            <div
              class="absolute top-1/2 inset-x-0 w-full border-b border-white/50"
            />
          </div>
          <div
            class="w-full flex-1 min-h-[2rem] border-b border-white/50 relative z-10"
            hour=""
          >
            <div
              class="text-white bg-[#212121] w-8 text-center absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
              text=""
            >
               9 
            </div>
            <div
              class="absolute top-1/2 inset-x-0 w-full border-b border-white/50"
            />
          </div>
          <div
            class="w-full flex-1 min-h-[2rem] border-b border-white/50 relative z-10"
            hour=""
          >
            <div
              class="text-white bg-[#212121] w-8 text-center absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
              text=""
            >
               10 
            </div>
            <div
              class="absolute top-1/2 inset-x-0 w-full border-b border-white/50"
            />
          </div>
          <div
            class="w-full flex-1 min-h-[2rem] border-b border-white/50 relative z-10"
            hour=""
          >
            <div
              class="text-white bg-[#212121] w-8 text-center absolute top-0 left-1/2 -translate-x-1/2 -translate-y-1/2"
              text=""
            >
               11 
            </div>
            <div
              class="absolute top-1/2 inset-x-0 w-full border-b border-white/50"
            />
          </div>
          
        </div>
      </div>
      <div
        class="flex-1 min-w-[30vw] flex flex-col items-center justify-center text-white opacity-60"
      >
        <p>
          No spaces have been selected
        </p>
      </div>
      
    </div>
  </div>
  
</div>
`;
